### DISCUSSION TITLE
Ordering PCB/pi hat?

### DISCUSSION THREAD

**Discussion by scoob8000** [nebhead/PiFire#58] [Q&A]
So I don't know if I'm missing something obvious but I can't figure out how to order the PCB from Easy Eda.   It's like it's just a bill of material that I can't do anything with after making an account and logging in.

Once you order is, ballpark what is the cost, and does it include the components (screw terminals, etc)?

I want to build one of these the worst way, if nothing else to monitor temps at first without any proprietary hardware.  I'm no stranger to doing stuff with pi's, basic electronics, 3d printing, etc.  But I'm just being dumb when it comes to getting a board.   (I'm too OCD to do all that jumpering with the GPIOs.)

**Comment by nebhead** [comment]
Ah, excellent question.  It's not intuitive to order the board via JLCPCB right now so let me see if I can step you through the process. 

1. Goto https://easyeda.com and login with our account (or register for an account and log in).  This account will be used with JLCPCB as well. 
2. Then, click the following link in the same browser (can be a different tab): [https://easyeda.com/nebhead77/pifire-v2](https://easyeda.com/nebhead77/pifire-v2)  
3. Next, scroll down to the picture of the PCB and click on the button labeled "Open in Editor"...
![image](https://user-images.githubusercontent.com/19876886/145314813-495f498c-31d2-44af-aaf5-c22f00142853.png)
4. This should open the EasyEDA editor with the PCB layout for PiFire.  Also, you should confirm that you are logged in as well (upper right hand corner. 
5. Now, you should be able to click on the "Fabrication" menu item and select "PCB Fabrication File (Gerber)". 
![image](https://user-images.githubusercontent.com/19876886/145315312-86b1a980-3f3c-450f-899b-923b4fe71709.png)
6. If it asks you to do a DRC check you can either skip or run the check.  Either way it should pass and take you to the PCB order screen.
7. Now you can select options for what you want to order (quantity and colors) and continue on to JLCPCB.com to order.  Note, this board wasn't designed for SMT, so don't even try to get the SMT option. :)  
![image](https://user-images.githubusercontent.com/19876886/145315705-8456c912-54af-4490-bbd6-9054f6d90308.png)

The rest should be self explanatory.  Keep most of the options at their default.  Opt out of SMT (it probably won't even be available).

**Reply by nebhead** [reply]
Oh, I forgot to mention that this is purely for the bare PCB.  You'll have to source things like headers, resistors and screw terminals elsewhere.  Amazon is a good source for some of this stuff, but can be obtained from Mouser, Digikey, Aliexpress, etc.. 

* Listing of some hardware components and sources listed in the documentation here: [https://nebhead.github.io/PiFire/hardware](https://nebhead.github.io/PiFire/hardware)
* Screw Terminals that I have used: [https://www.amazon.com/gp/product/B01F7KOM10](https://www.amazon.com/gp/product/B01F7KOM10)

Good luck and let us know how it goes if you get a chance to build this up!

**Comment by scoob8000** [comment]
Thanks!  I'll give the PCB order a go tomorrow.   I'm good with ordering the components and the assembly/soldering.   I've bought a lot from Mouser and Digikey in the past.   Hopefully I can get all the small stuff in one order from the same place.  That's the biggest pita.

I'll be tying this into my home automation and annoying the wife in no time.  :)    Alexa (every 5 minutes, when I'm not home): Meat temp is 122.  Meat temp is 123.   :)  :)

**Reply by nebhead** [reply]
>I'll be tying this into my home automation and annoying the wife in no time. :) Alexa (every 5 minutes, when I'm not home): Meat temp is 122. Meat temp is 123. :) :)

LOL... Yeah, still need to figure out how to get this connected to Home Assistant, but should be as easy as creating a template in HA config for the PiFire API.

**Reply by weberbox** [reply]
I just made the switch from smartthings to HA a few weeks ago. Still learning all the configs, yaml etc and irritating the wife lol. If I ever get around to learning HA enough hopefully I can get something going for it. I installed HACS and integrated my alexa's into the automation's. It is really cool that you can have alexa ask questions and then get a callback to turn off lights or lock doors etc. It would be cool to ask what the grill temp is or get notified when the meat hits your set temp etc. I am sure IFTTT can do that now but to get HA to control the grill it will need some more work. I started some googling to see if I could use the socket io to also control the grill from HA but I haven't gotten very far yet as I do not really understand how to make add-on's or whatever they are called for HA.

**Reply by cdcacrkeen** [reply]
So.... not sure if either of you guys are Discord users or not?  I'm not a gamer (my boys are), but I actually use Discord mostly for HA related things.  So head over to Digiblur's discord @ discord.digiblur.com and if you look through the section under "Other" you'll also note the #nothing-but-food and there's a subsection there that was just started for pifire smoker automation.  There's a bunch of really smart guys over there for HA and other things.  You'll probably feel right at home.  I'm already trying to resource the parts to put together a build of my own.  There's also a 3d-printing section there as well.  Hope to see you both over there!

**Reply by scoob8000** [reply]
Cool.   I'm not really on discord but I just joined and added that community.    I was a Smartthings user for years but moved over to Hubitat about a year ago.  Everything is local processing and so much faster.   The API is pretty simple too.  You can expose various controls to control via http requests.    I'm even using cameras in my dvr setup (blueiris) to trigger outside lighting.)

Just ordered the board.   Holy cow I didn't think it would be that cheap!   I assume these are made in china, hence the shipping cost..

Is the gpio header on a full blown pi the same as the pi zero?  I happen to have a unused pi laying around.   Might have to tweak the enclosure for it to fit.   i haven't even got that far to look at the design yet, but I'm excited.  :)

**Reply by cdcacrkeen** [reply]
You might take a strong look at HA.  I've got almost everything under local control on my setup, minus a few things like two Nest thermostats which I pull in with Homekit.  I am mostly running wifi light switches and bulbs that have been reflashed with Tasmota, which gives me instant results with local control.  I do have a few ESPHome devices too which I made a few years ago - a garage door opener which also reports the open/closed state of the door, and a water softener salt tank level monitor.  I also have 5 trees decorated with some 2812b LEDs that are running WLED, and a wreaths on the exterior doing the same.  Next year I'll entertain the permanently mounted house lights that do WLED.  Many of us are running blueiris and cameras as well.  Again, this stuff is right up your alley.  I remember looking at hubitat too, but wound up going HA as it was more appealing to me.  You might give it a look.  It has a huge community and gets frequent updates.

**Reply by nebhead** [reply]
I'm jumping on the Discord now!  :)

**Reply by scoob8000** [reply]
> You might take a strong look at HA. 

I'll have to revisit it.   I've got so much HA stuff, automations, graphing (influx/grafana) stuff going on I don't even want to think about changing hubs again.  lol

I'm running mostly off the shelf zwave/zigbee devices.  The only wifi stuff is tplink kasa gear (but with local control.   Gonna have to look into WLED.   :)

Glad I stumbled across this project.  Seems we're all a bunch of geeks that are also into cooking various stuff on the smoker.

**Reply by nebhead** [reply]
FYI: I've tweaked the PiFire API with the latest commit to the development branch (b9b91a9). The idea being that this can be used as a REST interface for Home Assistant. The /api/current call will provide current temps, setpoint temps, mode, etc. so we can use this information for a template to provide gauges, etc. And the for sending commands, we have POST /api/control to allow for rest_commands to be sent, or perhaps a REST switch. Anyway more research and testing required.

Found this article on the community forum that could be used to sketch out a basic interface, in a similar way: https://community.home-assistant.io/t/monitoring-a-bbq-with-ha-and-cloudbbq/217043

**Comment by scoob8000** [comment]
Sweet!   I just got my boards in from overseas.   Need to order the rest of
the parts.

**Comment by thompatry** [comment]
I just ordered all the parts. I am excited to see what I can contribute to the project.

**Reply by nebhead** [reply]
That's great!  Welcome to the club!
